docs/source/running-jobs/distributed-jobs.rst

```diff
@@ -187,31 +189,23 @@ To execute a distributed Ray program on many nodes, you can download the `traini
     workdir: .
 
     setup: |
-      conda activate ray
-      if [ $? -ne 0 ]; then
-        conda create -n ray python=3.10 -y
-        conda activate ray
-      fi
+      uv venv --python 3.10 --seed
+      source .venv/bin/activate
 
-      pip install "ray[train]"
-      pip install tqdm
-      pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
+      uv pip install "ray[train]" "click<8.2.0"
+      uv pip install tqdm
+      uv pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
 
     run: |
       sudo chmod 777 -R /var/tmp
-      HEAD_IP=`echo "$SKYPILOT_NODE_IPS" | head -n1`
-      if [ "$SKYPILOT_NODE_RANK" == "0" ]; then
-        ps aux | grep ray | grep 6379 &> /dev/null || ray start --head  --disable-usage-stats --port 6379
-        sleep 5
-        python train.py --num-workers $SKYPILOT_NUM_NODES
-      else
-        sleep 5
-        ps aux | grep ray | grep 6379 &> /dev/null || ray start --address $HEAD_IP:6379 --disable-usage-stats
-        # Add sleep to after `ray start` to give ray enough time to daemonize
-        sleep 5
-      fi
-
-.. warning::
+      source .venv/bin/activate
 
-  When using Ray, avoid calling ``ray stop`` as that will also cause the SkyPilot runtime to be stopped.
+      # This script is only available on skypilot-nightly>=1.0.0.dev20251113
+      # If you are using an older version, you can copy and paste the script from:
+      # https://github.com/skypilot-org/skypilot/blob/master/sky_templates/ray/start_cluster.sh
+      ~/sky_templates/ray/start_cluster.sh
```

Comment:
minor: curious, could we get rid of the `.sh` suffix. People may not need to understand it is a bash script. I suppose it should work if we have the `#!/bin/bash` header, even if we get rid of `.sh`

Reply:
Good point, I was thinking of that too earlier. I think it's a good idea, looks cleaner. Plus I imagine at some point it won't all just be bash scripts, we'll have python scripts too most likely. Which should also work if we add the `#!/usr/bin/env python` header.